src/viz/validation.ts

```diff
@@ -50,19 +50,19 @@ No files have been created.
 };
 
 export const parseBucketName = (bucketPath: string): string | undefined => {
-  const matches = bucketPath.match(/(gs:\/\/[^/\s]+)/);
+  const matches = bucketPath.match(/(gs:\/\/[-a-zA-Z]+)\/?([-a-zA-Z]+\/?)*$/);
   if (matches === null) {
     return undefined;
   }
-  return matches[0];
+  return matches[1];
 };
 
 export const hasBucketPermissions = async (
   gcsPath: string
 ): Promise<boolean | string> => {
   const gcsRootBucket = parseBucketName(gcsPath);
   if (gcsRootBucket === undefined) {
-    return `${gcsPath} is an invalid gcs bucket name.`;
+    throw new Error(`${gcsPath} is an invalid gcs bucket name.`);
```

Comment:
Would it be possible make `parseBucketName` throw this exception, rather than return an `undefined` sentinel value? I'd have a moderate preference in that direction. This would obviously require a change in the test to prefer listening for an exception, which is NBD.

Reply:
Sure, good suggestion.